perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java

```diff
@@ -249,7 +257,8 @@ public User getUserByExtSourceNameAndExtLogin(PerunSession sess, String extSourc
 		Utils.checkPerunSession(sess);
 
 		// Authorization
-		if(!AuthzResolver.isAuthorized(sess, Role.PERUNADMIN)) {
+		if(!AuthzResolver.isAuthorized(sess, Role.PERUNADMIN) &&
+				!AuthzResolver.isAuthorized(sess, Role.PERUNOBSERVER)) {
```

Comment:
You can drop PERUNADMIN check here. And also in all other such methods in this file please.